scripts/zkai_seq32_value_compatible_boundary_frontier_gate.py

```
@@ -0,0 +1,686 @@
+        if (artifact_id, path_text) != expected:
+            raise Seq32ValueCompatibleBoundaryFrontierError("source artifact inventory drift")
+        raw = read_repo_file(path, f"source artifact {artifact_id}")
+        if artifact.get("sha256") != sha256(raw):
+            raise Seq32ValueCompatibleBoundaryFrontierError("source artifact digest drift")
+        if artifact.get("size_bytes") != len(raw):
+            raise Seq32ValueCompatibleBoundaryFrontierError("source artifact size drift")
+
+    if has_mutation_metadata:
+        inventory = payload.get("mutation_inventory")
+        if not isinstance(inventory, dict):
+            raise Seq32ValueCompatibleBoundaryFrontierError("mutation inventory drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)